package/piksi_system_daemon/src/main.c

```diff
@@ -384,6 +393,85 @@ static int command_output(zloop_t *loop, zmq_pollitem_t *item, void *arg)
   return 0;
 }
 
+
+static u32 endian_flip_u32(u32 word)
+{
+  u32 b0,b1,b2,b3;
+
+  b0 = (word & 0x000000ff) << 24u;
+  b1 = (word & 0x0000ff00) << 8u;
+  b2 = (word & 0x00ff0000) >> 8u;
+  b3 = (word & 0xff000000) >> 24u;
+
+  return b0 | b1 | b2 | b3;
+}
+
+static void sbp_network_req(u16 sender_id, u8 len, u8 msg_[], void* context)
+{
+
+  msg_network_state_resp_t interfaces[SBP_MAX_NETWORK_INTERFACES];
+  memset(interfaces, 0, sizeof(interfaces));
+
+  struct ifaddrs *ifaddr, *ifa;
+  int family, s, n;
+  char host[NI_MAXHOST];
+
+  if (getifaddrs(&ifaddr) == -1) {
+      perror("getifaddrs");
+      return;
+  }
+
+  /* Walk through linked list, maintaining head pointer so we
+    can free list later */
+
+  for (ifa = ifaddr, n = 0; ifa != NULL; ifa = ifa->ifa_next, n++) {
+      if (ifa->ifa_addr == NULL)
+          continue;
+
+      if (strlen(ifa->ifa_name) >= sizeof(interfaces[0].interface_name)) {
+        perror("Interface name too long");
+        continue;
+      }
+
+      int interface_idx;
+      for (interface_idx = 0; interface_idx < SBP_MAX_NETWORK_INTERFACES; interface_idx++)
```

Comment:
Or alternatively just use a separate variable initialized outside the outer loop.